### PR TITLE
Update CronetFramework version number to 0.0.4

### PR DESCRIPTION
**Pull request by muxi**
Cronet.framework API is undergoing some substantial changes. The corresponding changes in gRPC are captured in #8684 by @gcasto. As a result I think we should advance the corresponding CocoaPods major version number and specify a new URL for download. New and old versions of Cronet.framework will coexist in cloud storage with unique URLs. The new version will be used for master, and the old version for grpc v1.0.x and any customer using current version of CronetFramework.podspec.

This PR is supposed to merge together with #8684.

Action items for landing the new interface:
- [x] Upload newly built Cronet.framework to cloud storage
- [x] Merge #8684 
- [x] Merge this PR
- [ ] Submit CronetFramework.podspec to CocoaPods repo

@gcasto - can we have a Cronet.framework that includes your changes?

### CODE REVIEW

File: src/objective-c/CronetFramework.podspec [hunk 1]
```diff
@@ -69,7 +69,7 @@ Pod::Spec.new do |s|
   s.vendored_framework = "Cronet.framework"
   s.author             = "The Chromium Authors"
   s.ios.deployment_target = "8.0"
-  s.source       = { :http => 'https://storage.googleapis.com/grpc-precompiled-binaries/cronet/Cronet.framework.zip' }
+  s.source       = { :http => 'https://storage.googleapis.com/grpc-precompiled-binaries/cronet/Cronet.framework-1.0.0.zip' }
```
Comment (jcanizales) [inline src/objective-c/CronetFramework.podspec]: I would put the version in a variable `v`, and use it both for `s.version` and here.

**Comment by muxi** [comment]
Yes I did mean different...

**Comment by muxi** [comment]
Ping~

File: src/objective-c/CronetFramework.podspec [hunk 2]
```diff
@@ -69,7 +70,8 @@ Pod::Spec.new do |s|
   s.vendored_framework = "Cronet.framework"
   s.author             = "The Chromium Authors"
   s.ios.deployment_target = "8.0"
-  s.source       = { :http => 'https://storage.googleapis.com/grpc-precompiled-binaries/cronet/Cronet.framework.zip' }
+  file =
```
Comment (jcanizales) [inline src/objective-c/CronetFramework.podspec]: I think you left this from a draft.

Comment (muxi) [inline src/objective-c/CronetFramework.podspec]: ehhhh...  hate myself

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
Test failures:
Bazel Full Build & Tests - #9379

**Pull request merged**